File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TreeViews_ViewTest.cpp
#include <gtest/gtest.h>
#include <memory>
#include "TreeViews_ViewModelTestEnvironment.h"
#include "TreeViews_ViewModel.h"
#include <string>
#include <vector>
#include "TreeViews_ViewModelMyTreeViewElementsRow.h"
#include "TreeViews_ViewModelMyStringRowHandlesRow.h"
#include "TreeViews_ViewModelMyParentIndexRowHandlesRow.h"
#include "TreeViews_ViewModelMyParentStringRowHandlesRow.h"
#include "TreeViews_ViewModelTestEnvironmentImpl.h"

namespace widgetassertions
{
  class TreeViews_ViewTest : public testing::Test
  {
  public:
    std::shared_ptr<TreeViews_ViewModel> sut;
    std::shared_ptr<TreeViews_ViewModelTestEnvironment> testEnvironment;
    virtual void BuildSut();
    virtual void given_empty_context();
    virtual void then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible();
    virtual void then_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1();
    virtual void then_MyParentIndexRowHandles_has_4_rows();
    virtual void then_MyParentStringRowHandles_has_4_rows();
    virtual void then_MyTreeViewElements_has_2_rows();
  protected:
    void SetUp() override;
  };
  void TreeViews_ViewTest::SetUp()
  {
    this->testEnvironment = std::make_shared<TreeViews_ViewModelTestEnvironmentImpl>();
    this->testEnvironment->Init();
  }
  TEST_F(TreeViews_ViewTest, My_Scenario_given_empty_context_when_then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible_and_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1_and_MyParentIndexRowHandles_has_4_rows_and_MyParentStringRowHandles_has_4_rows)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible();
    this->then_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1();
    this->then_MyParentIndexRowHandles_has_4_rows();
    this->then_MyParentStringRowHandles_has_4_rows();
  }
  TEST_F(TreeViews_ViewTest, TreeView_Ignore_Column_given_empty_context_when_then_MyTreeViewElements_has_2_rows)
  {
    this->given_empty_context();
    this->BuildSut();
    this->then_MyTreeViewElements_has_2_rows();
  }
  void TreeViews_ViewTest::BuildSut()
  {
    this->sut = this->testEnvironment->BuildSut();
  }
  void TreeViews_ViewTest::given_empty_context()
  {
  }
  void TreeViews_ViewTest::then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible()
  {
    auto& actualRows = this->sut->getMyTreeViewElementsTreeRows();
    ASSERT_EQ(2, actualRows.size()) << std::string("Expected that tree view MyTreeViewElements has 2 rows, but has ") + std::to_string(actualRows.size());
    // {
    auto& row0 = actualRows.at(0);
    ASSERT_EQ(0, row0->getRowIndex()) << std::string("Expected that tree view MyTreeViewElements row at index 0 has rowhandle <0>, but was <") + row0->getRowIndex() + std::string(">");
    ASSERT_EQ(0, row0->getRowDepth()) << std::string("Expected that tree view MyTreeViewElements row at index 0 has parent rowhandle <") + 0 + std::string(">, but was <") + row0->getRowDepth() + std::string(">");
    EXPECT_TRUE(row0->getIsIncludeCheckBoxChecked(), std::string("Expected that checkbox Include is checked, but was <") + row0->getIsIncludeCheckBoxChecked() + std::string(">"));
    ASSERT_EQ(std::string("A"), row0->getNameLabelText()) << std::string("Expected that label Name has text <") + std::string("A") + std::string(">, but was <") + row0->getNameLabelText() + std::string(">");
    ASSERT_EQ(std::string("B"), row0->getDescriptionLabelText()) << std::string("Expected that label Description has text <") + std::string("B") + std::string(">, but was <") + row0->getDescriptionLabelText() + std::string(">");
    ASSERT_EQ(std::string("image_triangle"), row0->getInfoImageName()) << std::string("Expected that image Info has ") + std::string("image <image_triangle>") + std::string(", but was <") + row0->getInfoImageName() + std::string(">");
    // }
    // {
    auto& row1 = actualRows.at(1);
    ASSERT_EQ(1, row1->getRowIndex()) << std::string("Expected that tree view MyTreeViewElements row at index 1 has rowhandle <1>, but was <") + row1->getRowIndex() + std::string(">");
    ASSERT_EQ(1, row1->getRowDepth()) << std::string("Expected that tree view MyTreeViewElements row at index 1 has parent rowhandle <") + 1 + std::string(">, but was <") + row1->getRowDepth() + std::string(">");
    EXPECT_FALSE(row1->getIsIncludeCheckBoxChecked(), std::string("Expected that checkbox Include is unchecked, but was <") + row1->getIsIncludeCheckBoxChecked() + std::string(">"));
    ASSERT_EQ(std::string("C"), row1->getNameLabelText()) << std::string("Expected that label Name has text <") + std::string("C") + std::string(">, but was <") + row1->getNameLabelText() + std::string(">");
    ASSERT_EQ(std::string("D"), row1->getDescriptionLabelText()) << std::string("Expected that label Description has text <") + std::string("D") + std::string(">, but was <") + row1->getDescriptionLabelText() + std::string(">");
    ASSERT_EQ(std::string("image_circle"), row1->getInfoImageName()) << std::string("Expected that image Info has ") + std::string("image <image_circle>") + std::string(", but was <") + row1->getInfoImageName() + std::string(">");
    // }
    ASSERT_EQ(1, this->sut->getMyTreeViewElementsTreeSelectedRow()) << std::string("Expected that tree view MyTreeViewElements has selected row with row handle <") + 1 + std::string(">, but was <") + this->sut->getMyTreeViewElementsTreeSelectedRow() + std::string(">");
    EXPECT_TRUE(this->sut->getIsMyTreeViewElementsTreeEnabled(), std::string("Expected tree view MyTreeViewElements is enabled, but it was disabled"));
    EXPECT_TRUE(this->sut->getIsMyTreeViewElementsTreeVisible(), std::string("Expected tree view MyTreeViewElements is visible, but it was hidden"));
    EXPECT_FALSE(this->sut->getIsMyTreeViewElementsTreeDescriptionColumnVisible(), std::string("Expected tree column Description is hidden, but it was visible"));
  }
  void TreeViews_ViewTest::then_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1()
  {
    auto& actualRows = this->sut->getMyStringRowHandlesTreeRows();
    ASSERT_EQ(2, actualRows.size()) << std::string("Expected that tree view MyStringRowHandles has 2 rows, but has ") + std::to_string(actualRows.size());
    // {
    auto& row0 = actualRows.at(0);
    ASSERT_EQ(std::string("ROW0"), row0->getRowHandle()) << std::string("Expected that tree view MyStringRowHandles row at index 0 has rowhandle <ROW0>, but was <") + row0->getRowHandle() + std::string(">");
    ASSERT_EQ(0, row0->getRowDepth()) << std::string("Expected that tree view MyStringRowHandles row at index 0 has parent rowhandle <") + 0 + std::string(">, but was <") + row0->getRowDepth() + std::string(">");
    ASSERT_EQ(std::string("A"), row0->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("A") + std::string(">, but was <") + row0->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row1 = actualRows.at(1);
    ASSERT_EQ(std::string("ROW1"), row1->getRowHandle()) << std::string("Expected that tree view MyStringRowHandles row at index 1 has rowhandle <ROW1>, but was <") + row1->getRowHandle() + std::string(">");
    ASSERT_EQ(0, row1->getRowDepth()) << std::string("Expected that tree view MyStringRowHandles row at index 1 has parent rowhandle <") + 0 + std::string(">, but was <") + row1->getRowDepth() + std::string(">");
    ASSERT_EQ(std::string("B"), row1->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("B") + std::string(">, but was <") + row1->getValuesLabelText() + std::string(">");
    // }
    ASSERT_EQ(std::string("ROW1"), this->sut->getMyStringRowHandlesTreeSelectedRow()) << std::string("Expected that tree view MyStringRowHandles has selected row with row handle <") + std::string("ROW1") + std::string(">, but was <") + this->sut->getMyStringRowHandlesTreeSelectedRow() + std::string(">");
  }
  void TreeViews_ViewTest::then_MyParentIndexRowHandles_has_4_rows()
  {
    auto& actualRows = this->sut->getMyParentIndexRowHandlesTreeRows();
    ASSERT_EQ(4, actualRows.size()) << std::string("Expected that tree view MyParentIndexRowHandles has 4 rows, but has ") + std::to_string(actualRows.size());
    // {
    auto& row0 = actualRows.at(0);
    ASSERT_EQ(0, row0->getRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 0 has rowhandle <0>, but was <") + row0->getRowIndex() + std::string(">");
    ASSERT_EQ(-1, row0->getParentRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 0 has parent rowhandle <") + -1 + std::string(">, but was <") + row0->getParentRowIndex() + std::string(">");
    ASSERT_EQ(std::string("A"), row0->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("A") + std::string(">, but was <") + row0->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row1 = actualRows.at(1);
    ASSERT_EQ(1, row1->getRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 1 has rowhandle <1>, but was <") + row1->getRowIndex() + std::string(">");
    ASSERT_EQ(0, row1->getParentRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 1 has parent rowhandle <") + 0 + std::string(">, but was <") + row1->getParentRowIndex() + std::string(">");
    ASSERT_EQ(std::string("A.B"), row1->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("A.B") + std::string(">, but was <") + row1->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row2 = actualRows.at(2);
    ASSERT_EQ(2, row2->getRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 2 has rowhandle <2>, but was <") + row2->getRowIndex() + std::string(">");
    ASSERT_EQ(-1, row2->getParentRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 2 has parent rowhandle <") + -1 + std::string(">, but was <") + row2->getParentRowIndex() + std::string(">");
    ASSERT_EQ(std::string("C"), row2->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("C") + std::string(">, but was <") + row2->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row3 = actualRows.at(3);
    ASSERT_EQ(3, row3->getRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 3 has rowhandle <3>, but was <") + row3->getRowIndex() + std::string(">");
    ASSERT_EQ(2, row3->getParentRowIndex()) << std::string("Expected that tree view MyParentIndexRowHandles row at index 3 has parent rowhandle <") + 2 + std::string(">, but was <") + row3->getParentRowIndex() + std::string(">");
    ASSERT_EQ(std::string("C.D"), row3->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("C.D") + std::string(">, but was <") + row3->getValuesLabelText() + std::string(">");
    // }
  }
  void TreeViews_ViewTest::then_MyParentStringRowHandles_has_4_rows()
  {
    auto& actualRows = this->sut->getMyParentStringRowHandlesTreeRows();
    ASSERT_EQ(4, actualRows.size()) << std::string("Expected that tree view MyParentStringRowHandles has 4 rows, but has ") + std::to_string(actualRows.size());
    // {
    auto& row0 = actualRows.at(0);
    ASSERT_EQ(std::string("HANDLE_A"), row0->getRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 0 has rowhandle <HANDLE_A>, but was <") + row0->getRowHandle() + std::string(">");
    ASSERT_EQ(std::string("-"), row0->getParentRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 0 has parent rowhandle <") + std::string("-") + std::string(">, but was <") + row0->getParentRowHandle() + std::string(">");
    ASSERT_EQ(std::string("A"), row0->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("A") + std::string(">, but was <") + row0->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row1 = actualRows.at(1);
    ASSERT_EQ(std::string("HANDLE_B"), row1->getRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 1 has rowhandle <HANDLE_B>, but was <") + row1->getRowHandle() + std::string(">");
    ASSERT_EQ(std::string("HANDLE_A"), row1->getParentRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 1 has parent rowhandle <") + std::string("HANDLE_A") + std::string(">, but was <") + row1->getParentRowHandle() + std::string(">");
    ASSERT_EQ(std::string("A.B"), row1->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("A.B") + std::string(">, but was <") + row1->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row2 = actualRows.at(2);
    ASSERT_EQ(std::string("HANDLE_C"), row2->getRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 2 has rowhandle <HANDLE_C>, but was <") + row2->getRowHandle() + std::string(">");
    ASSERT_EQ(std::string("-"), row2->getParentRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 2 has parent rowhandle <") + std::string("-") + std::string(">, but was <") + row2->getParentRowHandle() + std::string(">");
    ASSERT_EQ(std::string("C"), row2->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("C") + std::string(">, but was <") + row2->getValuesLabelText() + std::string(">");
    // }
    // {
    auto& row3 = actualRows.at(3);
    ASSERT_EQ(std::string("HANDLE_D"), row3->getRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 3 has rowhandle <HANDLE_D>, but was <") + row3->getRowHandle() + std::string(">");
    ASSERT_EQ(std::string("HANDLE_C"), row3->getParentRowHandle()) << std::string("Expected that tree view MyParentStringRowHandles row at index 3 has parent rowhandle <") + std::string("HANDLE_C") + std::string(">, but was <") + row3->getParentRowHandle() + std::string(">");
    ASSERT_EQ(std::string("C.D"), row3->getValuesLabelText()) << std::string("Expected that label Values has text <") + std::string("C.D") + std::string(">, but was <") + row3->getValuesLabelText() + std::string(">");
    // }
  }
  void TreeViews_ViewTest::then_MyTreeViewElements_has_2_rows()
  {
    auto& actualRows = this->sut->getMyTreeViewElementsTreeRows();
    ASSERT_EQ(2, actualRows.size()) << std::string("Expected that tree view MyTreeViewElements has 2 rows, but has ") + std::to_string(actualRows.size());
    // {
    auto& row0 = actualRows.at(0);
    ASSERT_EQ(0, row0->getRowIndex()) << std::string("Expected that tree view MyTreeViewElements row at index 0 has rowhandle <0>, but was <") + row0->getRowIndex() + std::string(">");
    ASSERT_EQ(0, row0->getRowDepth()) << std::string("Expected that tree view MyTreeViewElements row at index 0 has parent rowhandle <") + 0 + std::string(">, but was <") + row0->getRowDepth() + std::string(">");
    EXPECT_TRUE(row0->getIsIncludeCheckBoxChecked(), std::string("Expected that checkbox Include is checked, but was <") + row0->getIsIncludeCheckBoxChecked() + std::string(">"));
    // }
    // {
    auto& row1 = actualRows.at(1);
    ASSERT_EQ(1, row1->getRowIndex()) << std::string("Expected that tree view MyTreeViewElements row at index 1 has rowhandle <1>, but was <") + row1->getRowIndex() + std::string(">");
    ASSERT_EQ(1, row1->getRowDepth()) << std::string("Expected that tree view MyTreeViewElements row at index 1 has parent rowhandle <") + 1 + std::string(">, but was <") + row1->getRowDepth() + std::string(">");
    ASSERT_EQ(std::string("D"), row1->getDescriptionLabelText()) << std::string("Expected that label Description has text <") + std::string("D") + std::string(">, but was <") + row1->getDescriptionLabelText() + std::string(">");
    // }
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/widgetassertions/TreeViews_ViewTest.cs
/// <filename>
///     TreeViews_ViewTest.cs
/// </filename>
using widgetassertions;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Collections.Generic;
using System;

[Microsoft.VisualStudio.TestTools.UnitTesting.TestClass]
public class TreeViews_ViewTest
{
    private widgetassertions.TreeViews_ViewModel sut;

    private widgetassertions.TreeViews_ViewModelTestEnvironment testEnvironment;

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestInitialize]
    public void SetUp()
    {
        this.testEnvironment = new widgetassertions.TreeViews_ViewModelTestEnvironmentImpl();
        this.testEnvironment.Init();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void My_Scenario_given_empty_context_when_then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible_and_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1_and_MyParentIndexRowHandles_has_4_rows_and_MyParentStringRowHandles_has_4_rows()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible();
        this.then_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1();
        this.then_MyParentIndexRowHandles_has_4_rows();
        this.then_MyParentStringRowHandles_has_4_rows();
    }

    [Microsoft.VisualStudio.TestTools.UnitTesting.TestMethod]
    public void TreeView_Ignore_Column_given_empty_context_when_then_MyTreeViewElements_has_2_rows()
    {
        this.given_empty_context();
        this.BuildSut();
        this.then_MyTreeViewElements_has_2_rows();
    }

    protected virtual void BuildSut()
    {
        this.sut = this.testEnvironment.BuildSut();
    }

    public virtual void given_empty_context()
    {
    }

    public virtual void then_MyTreeViewElements_has_2_rows_and_selected_row_index_1_and_is_enabled_and_is_visible_and_Description_is_not_visible()
    {
        var actualRows = this.sut.getMyTreeViewElementsTreeRows();
        Assert.AreEqual(2, actualRows.Count, "Expected that tree view MyTreeViewElements has 2 rows, but has " + Convert.ToString(actualRows.Count));

        {
            var row0 = actualRows[0];
            Assert.AreEqual(0, row0.getRowIndex(), "Expected that tree view MyTreeViewElements row at index 0 has rowhandle <0>, but was <" + Convert.ToString(row0.getRowIndex()) + ">");
            Assert.AreEqual(0, row0.getRowDepth(), "Expected that tree view MyTreeViewElements row at index 0 has parent rowhandle <" + Convert.ToString(0) + ">, but was <" + Convert.ToString(row0.getRowDepth()) + ">");
            Assert.IsTrue(row0.getIsIncludeCheckBoxChecked(), "Expected that checkbox Include is checked, but was <" + (row0.getIsIncludeCheckBoxChecked() ? "checked" : "unchecked") + ">");
            Assert.AreEqual("A", row0.getNameLabelText(), "Expected that label Name has text <" + "A" + ">, but was <" + row0.getNameLabelText() + ">");
            Assert.AreEqual("B", row0.getDescriptionLabelText(), "Expected that label Description has text <" + "B" + ">, but was <" + row0.getDescriptionLabelText() + ">");
            Assert.AreEqual("image_triangle", row0.getInfoImageName(), "Expected that image Info has " + "image <image_triangle>" + ", but was <" + row0.getInfoImageName() + ">");
        }

        {
            var row1 = actualRows[1];
            Assert.AreEqual(1, row1.getRowIndex(), "Expected that tree view MyTreeViewElements row at index 1 has rowhandle <1>, but was <" + Convert.ToString(row1.getRowIndex()) + ">");
            Assert.AreEqual(1, row1.getRowDepth(), "Expected that tree view MyTreeViewElements row at index 1 has parent rowhandle <" + Convert.ToString(1) + ">, but was <" + Convert.ToString(row1.getRowDepth()) + ">");
            Assert.IsFalse(row1.getIsIncludeCheckBoxChecked(), "Expected that checkbox Include is unchecked, but was <" + (row1.getIsIncludeCheckBoxChecked() ? "checked" : "unchecked") + ">");
            Assert.AreEqual("C", row1.getNameLabelText(), "Expected that label Name has text <" + "C" + ">, but was <" + row1.getNameLabelText() + ">");
            Assert.AreEqual("D", row1.getDescriptionLabelText(), "Expected that label Description has text <" + "D" + ">, but was <" + row1.getDescriptionLabelText() + ">");
            Assert.AreEqual("image_circle", row1.getInfoImageName(), "Expected that image Info has " + "image <image_circle>" + ", but was <" + row1.getInfoImageName() + ">");
        }
        Assert.AreEqual(1, this.sut.getMyTreeViewElementsTreeSelectedRow(), "Expected that tree view MyTreeViewElements has selected row with row handle <" + Convert.ToString(1) + ">, but was <" + Convert.ToString(this.sut.getMyTreeViewElementsTreeSelectedRow()) + ">");
        Assert.IsTrue(this.sut.getIsMyTreeViewElementsTreeEnabled(), "Expected tree view MyTreeViewElements is enabled, but it was disabled");
        Assert.IsTrue(this.sut.getIsMyTreeViewElementsTreeVisible(), "Expected tree view MyTreeViewElements is visible, but it was hidden");
        Assert.IsFalse(this.sut.getIsMyTreeViewElementsTreeDescriptionColumnVisible(), "Expected tree column Description is hidden, but it was visible");
    }

    public virtual void then_MyStringRowHandles_has_2_rows_and_selected_row_handle_ROW1()
    {
        var actualRows = this.sut.getMyStringRowHandlesTreeRows();
        Assert.AreEqual(2, actualRows.Count, "Expected that tree view MyStringRowHandles has 2 rows, but has " + Convert.ToString(actualRows.Count));

        {
            var row0 = actualRows[0];
            Assert.AreEqual("ROW0", row0.getRowHandle(), "Expected that tree view MyStringRowHandles row at index 0 has rowhandle <ROW0>, but was <" + row0.getRowHandle() + ">");
            Assert.AreEqual(0, row0.getRowDepth(), "Expected that tree view MyStringRowHandles row at index 0 has parent rowhandle <" + Convert.ToString(0) + ">, but was <" + Convert.ToString(row0.getRowDepth()) + ">");
            Assert.AreEqual("A", row0.getValuesLabelText(), "Expected that label Values has text <" + "A" + ">, but was <" + row0.getValuesLabelText() + ">");
        }

        {
            var row1 = actualRows[1];
            Assert.AreEqual("ROW1", row1.getRowHandle(), "Expected that tree view MyStringRowHandles row at index 1 has rowhandle <ROW1>, but was <" + row1.getRowHandle() + ">");
            Assert.AreEqual(0, row1.getRowDepth(), "Expected that tree view MyStringRowHandles row at index 1 has parent rowhandle <" + Convert.ToString(0) + ">, but was <" + Convert.ToString(row1.getRowDepth()) + ">");
            Assert.AreEqual("B", row1.getValuesLabelText(), "Expected that label Values has text <" + "B" + ">, but was <" + row1.getValuesLabelText() + ">");
        }
        Assert.AreEqual("ROW1", this.sut.getMyStringRowHandlesTreeSelectedRow(), "Expected that tree view MyStringRowHandles has selected row with row handle <" + "ROW1" + ">, but was <" + this.sut.getMyStringRowHandlesTreeSelectedRow() + ">");
    }

    public virtual void then_MyParentIndexRowHandles_has_4_rows()
    {
        var actualRows = this.sut.getMyParentIndexRowHandlesTreeRows();
        Assert.AreEqual(4, actualRows.Count, "Expected that tree view MyParentIndexRowHandles has 4 rows, but has " + Convert.ToString(actualRows.Count));

        {
            var row0 = actualRows[0];
            Assert.AreEqual(0, row0.getRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 0 has rowhandle <0>, but was <" + Convert.ToString(row0.getRowIndex()) + ">");
            Assert.AreEqual(-1, row0.getParentRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 0 has parent rowhandle <" + Convert.ToString(-1) + ">, but was <" + Convert.ToString(row0.getParentRowIndex()) + ">");
            Assert.AreEqual("A", row0.getValuesLabelText(), "Expected that label Values has text <" + "A" + ">, but was <" + row0.getValuesLabelText() + ">");
        }

        {
            var row1 = actualRows[1];
            Assert.AreEqual(1, row1.getRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 1 has rowhandle <1>, but was <" + Convert.ToString(row1.getRowIndex()) + ">");
            Assert.AreEqual(0, row1.getParentRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 1 has parent rowhandle <" + Convert.ToString(0) + ">, but was <" + Convert.ToString(row1.getParentRowIndex()) + ">");
            Assert.AreEqual("A.B", row1.getValuesLabelText(), "Expected that label Values has text <" + "A.B" + ">, but was <" + row1.getValuesLabelText() + ">");
        }

        {
            var row2 = actualRows[2];
            Assert.AreEqual(2, row2.getRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 2 has rowhandle <2>, but was <" + Convert.ToString(row2.getRowIndex()) + ">");
            Assert.AreEqual(-1, row2.getParentRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 2 has parent rowhandle <" + Convert.ToString(-1) + ">, but was <" + Convert.ToString(row2.getParentRowIndex()) + ">");
            Assert.AreEqual("C", row2.getValuesLabelText(), "Expected that label Values has text <" + "C" + ">, but was <" + row2.getValuesLabelText() + ">");
        }

        {
            var row3 = actualRows[3];
            Assert.AreEqual(3, row3.getRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 3 has rowhandle <3>, but was <" + Convert.ToString(row3.getRowIndex()) + ">");
            Assert.AreEqual(2, row3.getParentRowIndex(), "Expected that tree view MyParentIndexRowHandles row at index 3 has parent rowhandle <" + Convert.ToString(2) + ">, but was <" + Convert.ToString(row3.getParentRowIndex()) + ">");
            Assert.AreEqual("C.D", row3.getValuesLabelText(), "Expected that label Values has text <" + "C.D" + ">, but was <" + row3.getValuesLabelText() + ">");
        }
    }

    public virtual void then_MyParentStringRowHandles_has_4_rows()
    {
        var actualRows = this.sut.getMyParentStringRowHandlesTreeRows();
        Assert.AreEqual(4, actualRows.Count, "Expected that tree view MyParentStringRowHandles has 4 rows, but has " + Convert.ToString(actualRows.Count));

        {
            var row0 = actualRows[0];
            Assert.AreEqual("HANDLE_A", row0.getRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 0 has rowhandle <HANDLE_A>, but was <" + row0.getRowHandle() + ">");
            Assert.AreEqual("-", row0.getParentRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 0 has parent rowhandle <" + "-" + ">, but was <" + row0.getParentRowHandle() + ">");
            Assert.AreEqual("A", row0.getValuesLabelText(), "Expected that label Values has text <" + "A" + ">, but was <" + row0.getValuesLabelText() + ">");
        }

        {
            var row1 = actualRows[1];
            Assert.AreEqual("HANDLE_B", row1.getRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 1 has rowhandle <HANDLE_B>, but was <" + row1.getRowHandle() + ">");
            Assert.AreEqual("HANDLE_A", row1.getParentRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 1 has parent rowhandle <" + "HANDLE_A" + ">, but was <" + row1.getParentRowHandle() + ">");
            Assert.AreEqual("A.B", row1.getValuesLabelText(), "Expected that label Values has text <" + "A.B" + ">, but was <" + row1.getValuesLabelText() + ">");
        }

        {
            var row2 = actualRows[2];
            Assert.AreEqual("HANDLE_C", row2.getRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 2 has rowhandle <HANDLE_C>, but was <" + row2.getRowHandle() + ">");
            Assert.AreEqual("-", row2.getParentRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 2 has parent rowhandle <" + "-" + ">, but was <" + row2.getParentRowHandle() + ">");
            Assert.AreEqual("C", row2.getValuesLabelText(), "Expected that label Values has text <" + "C" + ">, but was <" + row2.getValuesLabelText() + ">");
        }

        {
            var row3 = actualRows[3];
            Assert.AreEqual("HANDLE_D", row3.getRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 3 has rowhandle <HANDLE_D>, but was <" + row3.getRowHandle() + ">");
            Assert.AreEqual("HANDLE_C", row3.getParentRowHandle(), "Expected that tree view MyParentStringRowHandles row at index 3 has parent rowhandle <" + "HANDLE_C" + ">, but was <" + row3.getParentRowHandle() + ">");
            Assert.AreEqual("C.D", row3.getValuesLabelText(), "Expected that label Values has text <" + "C.D" + ">, but was <" + row3.getValuesLabelText() + ">");
        }
    }

    public virtual void then_MyTreeViewElements_has_2_rows()
    {
        var actualRows = this.sut.getMyTreeViewElementsTreeRows();
        Assert.AreEqual(2, actualRows.Count, "Expected that tree view MyTreeViewElements has 2 rows, but has " + Convert.ToString(actualRows.Count));

        {
            var row0 = actualRows[0];
            Assert.AreEqual(0, row0.getRowIndex(), "Expected that tree view MyTreeViewElements row at index 0 has rowhandle <0>, but was <" + Convert.ToString(row0.getRowIndex()) + ">");
            Assert.AreEqual(0, row0.getRowDepth(), "Expected that tree view MyTreeViewElements row at index 0 has parent rowhandle <" + Convert.ToString(0) + ">, but was <" + Convert.ToString(row0.getRowDepth()) + ">");
            Assert.IsTrue(row0.getIsIncludeCheckBoxChecked(), "Expected that checkbox Include is checked, but was <" + (row0.getIsIncludeCheckBoxChecked() ? "checked" : "unchecked") + ">");
        }

        {
            var row1 = actualRows[1];
            Assert.AreEqual(1, row1.getRowIndex(), "Expected that tree view MyTreeViewElements row at index 1 has rowhandle <1>, but was <" + Convert.ToString(row1.getRowIndex()) + ">");
            Assert.AreEqual(1, row1.getRowDepth(), "Expected that tree view MyTreeViewElements row at index 1 has parent rowhandle <" + Convert.ToString(1) + ">, but was <" + Convert.ToString(row1.getRowDepth()) + ">");
            Assert.AreEqual("D", row1.getDescriptionLabelText(), "Expected that label Description has text <" + "D" + ">, but was <" + row1.getDescriptionLabelText() + ">");
        }
    }
}
